Sources/BraveWallet/Panels/Signature Request/SignMessageRequestView.swift

```diff
@@ -0,0 +1,400 @@
+  let action: () -> Void
+  
+  @Environment(\.pixelLength) private var pixelLength
+  
+  var body: some View {
+    VStack(spacing: 8) {
+      if needsPilcrowFormatted {
+        Text("\(Image(systemName: "exclamationmark.triangle.fill")) \(Strings.Wallet.signMessageConsecutiveNewlineWarning)")
+          .font(.subheadline.weight(.medium))
+          .foregroundColor(Color(.braveLabel))
+          .multilineTextAlignment(.center)
+      }
+      if hasUnknownUnicode {
+        Text("\(Image(systemName: "exclamationmark.triangle.fill"))  \(Strings.Wallet.signMessageRequestUnknownUnicodeWarning)")
+          .font(.subheadline.weight(.medium))
+          .foregroundColor(Color(.braveLabel))
+          .multilineTextAlignment(.center)
+      }
+      Button {
+        action()
+      } label: {
+        Text(isShowingOriginalMessage ? Strings.Wallet.signMessageShowUnknownUnicode : Strings.Wallet.signMessageShowOriginalMessage)
+          .font(.subheadline)
+          .foregroundColor(Color(.braveBlurpleTint))
+      }
+    }
+    .padding(12)
+    .frame(maxWidth: .infinity)
+    .background(
+      Color(.braveWarningBackground)
+        .overlay(
+          RoundedRectangle(cornerRadius: 10, style: .continuous)
+            .strokeBorder(Color(.braveWarningBorder), style: StrokeStyle(lineWidth: pixelLength))
+        )
+        .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
+    )
+  }
+}
+
+/// View that displays the current index, total number of items and a `Next` button to move to next index.
+private struct NextIndexButton: View {
+  
+  let currentIndex: Int
+  let count: Int
+  let nextTapped: () -> Void
+  
+  var body: some View {
+    HStack {
+      Text(String.localizedStringWithFormat(Strings.Wallet.transactionCount, currentIndex + 1, count))
+        .fontWeight(.semibold)
+      Button(action: {
+        nextTapped()
+      }) {
+        Text(Strings.Wallet.next)
+          .fontWeight(.semibold)
+          .foregroundColor(Color(.braveBlurpleTint))
+      }
+    }
+  }
+}
+
+extension String {
+  var hasUnknownUnicode: Bool {
+    // same requirement as desktop. Valid: [0, 127]
+    for c in unicodeScalars {
+      let ci = Int(c.value)
+      if ci > 127 {
+        return true
+      }
+    }
+    return false
+  }
+  
+  var hasConsecutiveNewLines: Bool {
+    // return true if string has two or more consecutive newline chars
+    return range(of: "\\n{3,}", options: .regularExpression) != nil
+  }
+  
+  var printableWithUnknownUnicode: String {
+    var result = ""
+    for c in unicodeScalars {
+      let ci = Int(c.value)
+      if let unicodeScalar = Unicode.Scalar(ci) {
+        if ci == 10 { // will keep newline char as it is
+          result += "\n"
+        } else {
+          // ascii char will be displayed as it is
+          // unknown (> 127) will be displayed as hex-encoded
+          result += unicodeScalar.escaped(asASCII: true)
+        }
+      }
+    }
+    return result
+  }
+}
```

Comment:
If these are specific to this view make these `fileprivate`, otherwise probably best to move them a separate file and add some unit tests